src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
-export interface Figure {
-
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || this.a >= this.b + this.c
+      || this.b >= this.a + this.c
+      || this.c >= this.a + this.b) {
+      throw new Error('Wrong triangle parameters');
+    }
+  }
+
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+    let area: number = Math.sqrt(s * (s - this.a)
+      * (s - this.b)
+      * (s - this.c));
+
+    area = Math.floor(area * 100) / 100;
 
+    return area;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = Shape.circle;
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('sides 1, 2 and 3 can\'t form a triangle');
```

Comment:
It's a circle :) Change the error message

Reply:
yes indeed. Replaced